### PR TITLE
added Login endpoints and modified Main.tsx

### PR DESCRIPTION
**Pull request by nimendoza**
## Proposed Changes

  - added LoginPage, LogoutPage, LoginTest
  - modified Main.tsx
  - installed axios and use-async-effect packages

### CODE REVIEW

File: web-client/src/pages/session/test.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    )
+  }
+
+  try {
+    const user = JSON.parse(userJson);
+    (async () => await axios.get(`http://localhost:4000/v1/users/${user.id}`, {
+      headers: {
+        'Authorization': jwt
+      }
+    }));
```
Comment (jabbawookiees) [inline web-client/src/pages/session/test.tsx:27]: I don't think this is needed anymore. This seems like a function definition that doesn't get called. Nor do you need the result anyway.

Comment (jabbawookiees) [reply web-client/src/pages/session/test.tsx:27]: Also, I don't think React components can `await` inside them. Correct me if I'm wrong, but isn't the desired effect you want something like:

Step 1: Render a loading screen and start checking if you're logged in. This second bit requires some async stuff and might take a few seconds if the internet is slow.

Step 2: Depending on the response, you want to replace the loading screen with either "You are logged in as X" or "You are logged out".

If so may I suggest some code that roughly looks like this instead:

```
const LoginTest = () => {
  const getCurrentUser = async () => {
    // Do the thing that gets the JWT and makes a server thing.
    // return User if logged in, returns null if logged out
  }

  [status, setStatus] = useState('loading');
  [user, setUser] = useState(null);

  const checkIfLoggedInAndUpdateStatus = async () => {
    const user = await getCurrentUser();
    if (user != null) {
      setStatus('logged-in');
      setUser(user);
    } else {
      setStatus('logged-out');
    }
  }
  useAsyncEffect(checkIfLoggedInAndUpdateStatus);
  if (status == 'loading') {
    return <div>Loading</div>
  } else if (status == 'logged-in') {
    return <div>You are logged in as {user.email}</div>
  } else if (status == 'logged-out') {
    return <div>You are logged out</div>
  }
};

export default LoginTest;
```

**Review by jabbawookiees — CHANGES_REQUESTED** [review]

File: web-client/package.json [hunk 2]
```diff
@@ -26,7 +27,8 @@
     "remark-parse": "^10.0.1",
     "remark-rehype": "^10.1.0",
     "styled-jsx-plugin-postcss": "^4.0.1",
-    "unified": "^10.1.1"
+    "unified": "^10.1.1",
```
Comment (jabbawookiees) [inline web-client/package.json:30]: Also not sure what this package is for. Does this do anything? Can we remove it?

Comment (nimendoza) [reply web-client/package.json:30]: Apparently it's used in one of the boilerplate pages. (https://github.com/noiph/hurado/blob/main/web-client/src/utils/Markdown.ts)

**Pull request merged**